src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,62 @@
 'use strict';
 
+// Animal class - represents the general characteristics of all animals
 class Animal {
-  // write your code here
+  constructor(name) {
+    // The name property is the name of the animal
+    this.name = name;
+    // The health property is the animal's health, the default is 100
+    this.health = 100;
+    // Adds the current Animal object to the alive array,
+    // which displays all living animals
+    Animal.alive.push(this);
+  }
 }
 
+// Static array alive - stores all living animals,
+//  is common to all instances of the Animal class
+Animal.alive = [];
+
+// The Herbivore class represents a herbivore animals
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    // Property hidden - indicates whether the herbivore
+    // is hidden (default - false)
+    this.hidden = false;
+  }
+
+  // The hide method hides the animal by changing the hidden property to true
+  hide() {
+    this.hidden = true;
+  }
+
+  // checkHealth method - checks the health of the herbivore
+  // and removes it from the alive array if health <= 0
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
the method of deleting items from static property - is better to make static as well.
and better to use filter method here.